src/main/kotlin/com/zepben/auth/client/ZepbenTokenFetcher.kt

```diff
@@ -229,3 +261,66 @@ fun createTokenFetcher(
     }
     return null
 }
+
+/**
+ * Helper method to fetch auth related configuration from `confAddress` and create a `ZepbenTokenFetcher`
+ *
+ * @param confAddress Location to retrieve authentication configuration from. Must be a HTTP address that returns a JSON response.
+ * @param authTypeField The field name to look up in the JSON response from the confAddress for `tokenFetcher.authMethod`.
+ * @param audienceField The field name to look up in the JSON response from the confAddress for `tokenFetcher.authMethod`.
+ * @param issuerDomainField The field name to look up in the JSON response from the confAddress for `tokenFetcher.authMethod`.
+ * @param verifyCertificates: Whether to verify the certificate when making HTTPS requests. Note you should only use a trusted server
+ *                            and never set this to False in a production environment.
+ *
+ * @returns: A `ZepbenTokenFetcher` if the server reported authentication was configured, otherwise None.
+ */
+fun createTokenFetcher(
+    confAddress: String,
+    authTypeField: String = "authType",
+    audienceField: String = "audience",
+    issuerDomainField: String = "issuer",
+    verifyCertificates: Boolean
+) = createTokenFetcher(
+    confAddress,
+    authTypeField,
+    audienceField,
+    issuerDomainField,
+    confClient = HttpClient.newBuilder().sslContext(
+        if (verifyCertificates) SSLContext.getDefault() else SSLContextUtils.allTrustingSSLContext()
+    ).build(),
+    authClient = HttpClient.newBuilder().sslContext(
+        if (verifyCertificates) SSLContext.getDefault() else SSLContextUtils.allTrustingSSLContext()
+    ).build()
+)
+
+/**
+ * Helper method to fetch auth related configuration from `confAddress` and create a `ZepbenTokenFetcher`
+ *
+ * @param confAddress Location to retrieve authentication configuration from. Must be a HTTP address that returns a JSON response.
+ * @param confCAFilename Filename of X.509 CA certificate used to verify HTTPS responses from configuration service. Leave as null to use system CAs.
+ * @param authCAFilename Filename of X.509 CA certificate used to verify HTTPS responses from token service. Leave as null to use system CAs.
+ * @param authTypeField The field name to look up in the JSON response from the confAddress for `tokenFetcher.authMethod`.
+ * @param audienceField The field name to look up in the JSON response from the confAddress for `tokenFetcher.authMethod`.
+ * @param issuerDomainField The field name to look up in the JSON response from the confAddress for `tokenFetcher.authMethod`.
+ *
+ * @returns: A `ZepbenTokenFetcher` if the server reported authentication was configured, otherwise None.
+ */
+fun createTokenFetcher(
```

Comment:
Same here as mentioned above - but make confCAFilename and authCAFilename not nullable.

Reply:
I remember why I did this: For the case where the user only wants to specify `confCAFilename` and not `authCAFilename`.